### PR TITLE
Add explicit parameters for AIOHttpConnection and AsyncTransport

### LINKED ISSUE #1405: Add 'timeout' parameter explicitly to AIOHttpConnection.__init__()
It's in the docstring but gets passed transparently via `**kwargs` which makes discovery for users tougher.

### PR DESCRIPTION
**Pull request by V1NAY8**
Closes #1405 

@sethmlarson  take a look at this. Let me know if any changes are required.

Also synchronized `AsyncTransport` , `Transport` =>  `__init__()`

### CODE REVIEW

**Comment by V1NAY8** [comment]
I will fix the tests, probably should pass keyword arguments

File: elasticsearch/_async/transport.py [hunk 1]
```diff
@@ -44,7 +46,25 @@ class AsyncTransport(Transport):
...
+        sniffer_timeout=None,
+        sniff_timeout=0.1,
+        sniff_on_connection_fail=False,
+        serializer=JSONSerializer(),
+        serializers=None,
+        default_mimetype="application/json",
+        max_retries=3,
+        retry_on_status=(502, 503, 504),
+        retry_on_timeout=False,
+        send_get_body_as="GET",
```
Comment (sethmlarson) [inline elasticsearch/_async/transport.py:65]: This is missing a single parameter `meta_header`, can you also add this parameter in `elasticsearch/_async/transport.pyi`?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for this, almost there! Only the `meta_header` parameter missing.

**Review by sethmlarson — APPROVED** [review]
LGTM, thanks for this! :muscle:

**Comment by sethmlarson** [comment]
jenkins test this please

**Pull request merged**